.github/workflows/publish-helm-charts.yaml

```diff
@@ -0,0 +1,122 @@
+name: publish-helm-chart
+
+on:
+  push:
+  workflow_dispatch:
```

Comment:
@adity1raut @kubeflow/kubeflow-trainer-team @kramaranya @szaher Shall we run this script when we publish a new tag, like we do to publish images: https://github.com/kubeflow/trainer/blob/master/.github/workflows/build-and-push-images.yaml#L8-L9

I would imagine for now, we should publish versions of charts similar to images:
For the master branch
```
ghcr.io/kubeflow/charts/kubeflow-trainer:latest
ghcr.io/kubeflow/charts/kubeflow-trainer:sha-6109d36
```

For the releases:
```
ghcr.io/kubeflow/charts/kubeflow-trainer:2.1.0
```


WDYT ?